src/metabase/upload.clj

```diff
@@ -300,24 +343,29 @@
     - ::text
     - ::date
     - ::datetime
+    - ::pk
 
   A column that is completely blank is assumed to be of type ::text."
-  [csv-file]
+  [driver csv-file]
   (with-open [reader (bom/bom-reader csv-file)]
     (let [[header & rows] (csv/read-csv reader)]
-      (rows->schema header (sample-rows rows)))))
+      (rows->schema driver header (sample-rows rows)))))
 
 (defn load-from-csv!
   "Loads a table from a CSV file. If the table already exists, it will throw an error.
    Returns the file size, number of rows, and number of columns."
   [driver db-id table-name ^File csv-file]
-  (let [col->upload-type   (detect-schema csv-file)
-        col->database-type (update-vals col->upload-type (partial driver/upload-type->database-type driver))
-        column-names       (keys col->upload-type)]
-    (driver/create-table! driver db-id table-name col->database-type)
+  (let [col->upload-type               (detect-schema driver csv-file)
+        col->column-spec               (update-vals col->upload-type (partial upload-type->column-spec driver))
+        columns-to-insert->upload-type (->> col->upload-type
```

Comment:
For consistency you can use `col` instead of `column`. And by convention maps like this should be named like `key->value` not `keys->val`. Because `keys->val` could be read as a map where the keys are sequences.

```suggestion
        col-to-insert->upload-type (->> col->upload-type
```